rust/actyx/swarm/harness/src/bin/versions.rs

```diff
@@ -94,52 +86,70 @@ fn main() -> anyhow::Result<()> {
                 m.peer_id()
             );
         }
+        Ok(())
+    }
 
-        let machines = sim.machines().iter().map(|m| m.id()).collect::<Vec<_>>();
-
-        let started = Instant::now();
-        for i in &machines {
+    async fn ensure_all_machines_are_up(
+        sim: &mut Netsim<Command, Event>,
+        machine_ids: &Vec<MachineId>,
+    ) -> anyhow::Result<()> {
+        for i in machine_ids {
             let ip = sim.machine(*i).addr();
             let ns = sim.machine(*i).namespace();
             loop {
-                let req = std::thread::spawn(move || {
+                let req = {
```

Comment:
This changes semantics: `ns.enter()` places the current thread into the `ns` namespace, hence the new thread.

Reply:
I suspected something like that was happening. So I guess that this works instead:
```rust
let req = timeout(Duration::from_secs(10), async { 
    ns.enter().unwrap();
    reqwest::get(format!("http://{}:4454/api/v2/node/id", ip))
}).await;
```

Reply:
no: now you put some arbitrary tokio thread into that namespace …

Reply:
Ah right, the thread doesn't leave the namespace when finished. I'll get the thread back